### PR TITLE
Fixed execution with static configuration width and floating-point un…

### PR DESCRIPTION
**Pull request by Mohit1295**
…it. This commit does not support atomic or fused-multiply-add instructions

### CODE REVIEW

File: testbenches/debugPrints.sv [hunk 1]
```diff
@@ -2123,7 +2136,7 @@ task alist_debug_print;
     
     $fwrite(fd_alist, "\nexceptionFlag:");
     for (i = 0; i < `COMMIT_WIDTH; i++)
-        $fwrite(fd_alist, "       %b ", coreTop.activeList.exceptionFlag[i]);
+        $fwrite(fd_alist, "       %b ", coreTop.activeList.exceptionFlag[i]);
```
Comment (anycore) [inline testbenches/debugPrints.sv]: Remove whitespaces like this - unnecessary diffs

File: retire/ActiveList.sv [hunk 2]
```diff
@@ -767,6 +769,65 @@ ALEXCPT_RAM #(
 	);
 
 
+/*-----------------------Changes: Mohit----------------------*/
+// RAM instance for Floating-point Exception bits corresponding to all FP 
+// instructions. The exception bits are then updated in CSR_FFLAGS during retire
+ALEXCPT_RAM #(
+  .RPORT      (`COMMIT_WIDTH),
+  .WPORT      (1),
```
Comment (anycore) [inline retire/ActiveList.sv:777]: Tab missing??

File: retire/ActiveList.sv [hunk 3]
```diff
@@ -329,49 +331,49 @@ ALDATA_RAM #(
 
 
 	.addr0wr_i (bistEn ? bistAddrWr : tailAddr[0]),
-  .we0_i     (bistEn ? 1'b1       : alPacket_i[0].valid),
+  .we0_i     (bistEn ? 1'b1       : alPacket_i[0].valid),
```
Comment (anycore) [inline retire/ActiveList.sv]: whitespace after line end - unnecessary diff

File: regRead/SupRegFile.sv [hunk 4]
```diff
@@ -234,6 +241,41 @@ begin
     regWrAddrCommit <=  regWrAddr_i;
     regWrDataCommit <=  regWrData_i;
   end
+  //Changes: Mohit (Else if block added to handle case when misprediction occurs
+  //since execution results of CSRs need to flushed)
+  else if(flush_i) begin
```
Comment (anycore) [inline regRead/SupRegFile.sv]: Why is this necessary? If the instruction never commits, its regWrDataCommit and regWrAddrCommit needs to be thrown away. If that is not happening correctly, change needs to be made in the block below, not here. Doing what you have done now adds a lot of reads from the CSR on every flush. Functionally, this is ok but is an inefficient microarchitecture.

File: regRead/SupRegFile.sv [hunk 5]
```diff
@@ -350,9 +392,11 @@ begin
   // Write the register when the CSR instruction commits
   else
   begin
```
Comment (anycore) [inline regRead/SupRegFile.sv:380]: This is the block that should be changed to prevent writing a flushed value to the actual architectural CSR.

File: execute/ExecutionPipe_SC.sv [hunk 6]
```diff
@@ -19,9 +19,9 @@
 `timescale 1ns/100ps
 
 module ExecutionPipe_SC #(
-	parameter  SIMPLE  = `PIPE_HAS_SIMPLE,
-	parameter  COMPLEX = `PIPE_HAS_COMPLEX,
-  parameter  FP      = `PIPE_HAS_FP
+	parameter  SIMPLE  = 1,//`PIPE_HAS_SIMPLE,
```
Comment (anycore) [inline execute/ExecutionPipe_SC.sv]: Why is this hardcoded?

File: execute/Execute_Ctrl.sv [hunk 7]
```diff
@@ -89,7 +96,7 @@ ForwardCheck src1Bypass (
 
 	.bypassPacket_i                     (bypassPacket_i),
 
-	.dataOut_o                          (src1Data)
+	.dataOut_o                          (src1Data)
```
Comment (anycore) [inline execute/Execute_Ctrl.sv]: Whitespace at line end - unnecessary diff

File: decode/Decode.sv [hunk 8]
```diff
@@ -31,7 +31,7 @@ module Decode(
 	input                             fs2Ready_i,
 
 	input  decPkt                     decPacket_i [0:`FETCH_WIDTH-1],
-
+
```
Comment (anycore) [inline decode/Decode.sv]: unnecessary whitespace.

File: regRead/SupRegFile.sv (hunk 4 above)
Comment (Mohit1295) [reply regRead/SupRegFile.sv]: As I mentioned in my report, the bug I specifically encountered was that the FENCE instruction which causes a prior flushed CSR instruction to commit its result since commitReg_i becomes 1. So then the FENCE and all other CSR instructions that currently do not change  regWrDataCommit and regWrAddrCommit would have to update those registers during their execution to avoid this issue. Instead I implemented this loopback mechanism which I do understand is inefficient microarchitecture.

File: regRead/SupRegFile.sv (hunk 5 above)
Comment (Mohit1295) [reply regRead/SupRegFile.sv:380]: How will the core be able to keep track of a flushed CSR instruction? What if the CSR we tried to update in bad path of mispredicted branch is the same in good path?

File: execute/ExecutionPipe_SC.sv (hunk 6 above)
Comment (Mohit1295) [reply execute/ExecutionPipe_SC.sv]: I had made that change to understand parameter passing from the config-files. I forgot to change it back. Irrespective of that the parameters are overridden during instantiation in Core_OOO.sv. But I will revert them back to the way they were.

File: retire/ActiveList.sv (hunk 2 above)
Comment (Mohit1295) [reply retire/ActiveList.sv:777]: Can you elaborate on this comment? I do not understand its meaning.

File: regRead/SupRegFile.sv (hunk 4 above)
Comment (anycore) [reply regRead/SupRegFile.sv]: I understand. Do the following instead:

Add a regWrValid bit that is set when a CSR instruction is executed and is reset on a flush. Then when commit goes high, check if the valid bit is set and only generate write enables if it is. This should take care of wrong path CSR writes propagating after a misprediction.  I can test the patch.

Comment (Mohit1295) [reply regRead/SupRegFile.sv]: Can you check my latest commit to confirm that the implementation is what you expect?

File: regRead/SupRegFile.sv [hunk 9]
```diff
@@ -240,41 +241,12 @@ begin
...
     regWrAddrCommit <=  regWrAddr_i;
     regWrDataCommit <=  regWrData_i;
+    regWrValid <=  1'b1;
```
Comment (anycore) [inline regRead/SupRegFile.sv]: This needs a reset value. I recommend separating this out to a different always_ff block. Set it to 1'b1 on a write and set it to 1'b0 on reset and flush. The implementation looks good otherwise.

Comment (anycore) [reply regRead/SupRegFile.sv]: Also, make sure FENCE does not generate a regWrEn_i.

Comment (Mohit1295) [reply regRead/SupRegFile.sv]: I replaced the latest commit with the suggested implementation. FENCE is identified as OP_MISC_MEM instruction for which the regWrEn_i is not set. Please review the code.

**Review by anycore — APPROVED** [review]
